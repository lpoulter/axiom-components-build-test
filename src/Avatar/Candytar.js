import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import { colors } from '@brandwatch/axiom-materials';
import renderFilter from './_filter';
import './Candytar.css';

const availableColors = Object.keys(colors.userDefinedGraphColors);
const defaultPickerFn = (colors) =>
  colors[Math.floor(Math.random() * colors.length)];

export default class Candytar extends Component {
  static propTypes = {
    /** Color of the Candytar */
    color: PropTypes.oneOf([
      'tiny-clanger',
      'critical-mass',
      'paradise-lost',
      'serene-sea',
      'giant-leap',
      'moon-lagoon',
      'terra-form',
      'primeval-soup',
      'new-horizon',
      'blast-off',
      'ground-control',
      'luna-dust',
    ]),
    /**
     * The User's initials to appear in the candytar.
     * If no initials are supplied the silhouette will be shown by default.
     */
    initials: PropTypes.string,
    /**
     * A function that can be passed to determine the colour of the candytar.
     * By default if no colour is provided it will randomly picker a colour
     * from the available options. Given function is called with an array of
     * the available colors.
     */
    picker: PropTypes.func,
    /** Size of the Candytar */
    size: PropTypes.string.isRequired,
  };

  static defaultProps = {
    picker: defaultPickerFn,
    initials: '',
  };

  componentDidMount = renderFilter

  render() {
    const { picker, color = picker(availableColors), size, initials } = this.props;
    const style = { height: size, width: size };
    const classes = classnames('ax-candytar', `ax-candytar--${color}`);
    const userInitials = initials.slice(0, 2).toUpperCase();

    /* eslint-disable max-len */
    const showSilhouette = (
      <path
          className="ax-candytar__body"
          d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
          filter="url(#ax-candytar__filter)">
      </path>
    );

    const showInitials = (
      <text className="ax-candytar__initials" textAnchor="middle" x="50%" y="65%">{ userInitials }</text>
    );
    /* eslint-enable max-len */

    return (
      <svg className={ classes } style={ style } viewBox="0 0 32 32">
        <circle className="ax-candytar__background" cx="16" cy="16" r="16"></circle>
        { initials ? showInitials : showSilhouette }
      </svg>
    );
  }
}
