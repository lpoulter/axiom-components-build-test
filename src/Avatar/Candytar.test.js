import React from 'react';
import renderer from 'react-test-renderer';
import { colors } from '@brandwatch/axiom-materials';
import Candytar from './Candytar';

const getComponent = (props = {}) =>
  renderer.create(
    <Candytar size="2rem" src="/image/path" { ...props } />
  );

describe('Candytar', () => {
  it('renders with custom size', () => {
    const component = getComponent({ color: 'tiny-clanger', size: '4rem' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with color', () => {
    Object.keys(colors.userDefinedGraphColors).forEach((color) => {
      it(color, () => {
        const component = getComponent({ color });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
