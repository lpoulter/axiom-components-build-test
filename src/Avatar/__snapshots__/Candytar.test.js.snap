// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Candytar renders with color new-horizon 1`] = `
<svg
  className="ax-candytar ax-candytar--new-horizon"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with color giant-leap 1`] = `
<svg
  className="ax-candytar ax-candytar--giant-leap"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with color ground-control 1`] = `
<svg
  className="ax-candytar ax-candytar--ground-control"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with color primeval-soup 1`] = `
<svg
  className="ax-candytar ax-candytar--primeval-soup"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with color terra-form 1`] = `
<svg
  className="ax-candytar ax-candytar--terra-form"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with color luna-dust 1`] = `
<svg
  className="ax-candytar ax-candytar--luna-dust"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with color serene-sea 1`] = `
<svg
  className="ax-candytar ax-candytar--serene-sea"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with color blast-off 1`] = `
<svg
  className="ax-candytar ax-candytar--blast-off"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with color critical-mass 1`] = `
<svg
  className="ax-candytar ax-candytar--critical-mass"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with color paradise-lost 1`] = `
<svg
  className="ax-candytar ax-candytar--paradise-lost"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with color tiny-clanger 1`] = `
<svg
  className="ax-candytar ax-candytar--tiny-clanger"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with color moon-lagoon 1`] = `
<svg
  className="ax-candytar ax-candytar--moon-lagoon"
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;

exports[`Candytar renders with custom size 1`] = `
<svg
  className="ax-candytar ax-candytar--tiny-clanger"
  style={
    Object {
      "height": "4rem",
      "width": "4rem",
    }
  }
  viewBox="0 0 32 32"
>
  <circle
    className="ax-candytar__background"
    cx="16"
    cy="16"
    r="16"
  />
  <path
    className="ax-candytar__body"
    d="M16 20c-4 0-7-3-7-7s3-7 7-7 7 3 7 7-3 7-7 7zM6 28.5c1.7-3.8 5.6-6.5 10-6.5 4.5 0 8.3 2.7 10 6.5C23.4 30.7 20 32 16 32s-7.3-1.3-10-3.5z"
    filter="url(#ax-candytar__filter)"
  />
</svg>
`;
